frontend/packages/topology/src/components/quick-search/QuickSearchModalBody.tsx

```diff
@@ -132,6 +138,7 @@ const QuickSearchModalBody: React.FC<QuickSearchModalBodyProps> = ({
 
     const onOutsideClick = (e: MouseEvent) => {
       if (!ref.current?.contains(e.target as Node)) {
+        removeQueryArgument('catalogSearch');
```

Comment:
You should only remove the query argument if there's no search term. We want to save the query argument if user clicks outside in the middle of search and comes back to the search.

Reply:
We are already removing query argument when there is no search term.

Reply:
@abhinandan13jan as @rohitkrai03 mentioned don't remove query argument on outside click since the user should be able to come back to their incomplete search after the modal got closed on outside click.

Reply:
@abhinandan13jan ^^

Reply:
updated